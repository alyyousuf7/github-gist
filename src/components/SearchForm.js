import { useState } from 'react';
import PropTypes from 'prop-types';

import './SearchForm.css';

function SearchForm({ onSubmit }) {
  const [username, setUsername] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    if (onSubmit) {
      onSubmit(username);
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input id="username" type="text" placeholder="username" value={username} onChange={(e) => setUsername(e.target.value)} />
      <input type="submit" value="search" />
    </form>
  );
}

SearchForm.propTypes = {
  onSubmit: PropTypes.func,
}

export default SearchForm;
