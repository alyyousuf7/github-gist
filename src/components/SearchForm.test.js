import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event';

import SearchForm from './SearchForm';

describe('SearchForm', () => {
  it('should render correctly', () => {
    const onSubmit = jest.fn();
    const { container } = render(<SearchForm onSubmit={onSubmit} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call onSubmit with username', async () => {
    const onSubmit = jest.fn();
    const { container } = render(<SearchForm onSubmit={onSubmit} />);

    userEvent.type(container.querySelector('#username'), 'some username');
    userEvent.click(container.querySelector('input[type=submit]'));

    expect(onSubmit).toBeCalledWith('some username');
  });
});
