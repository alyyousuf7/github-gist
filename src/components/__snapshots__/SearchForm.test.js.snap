// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should render correctly 1`] = `
<form>
  <input
    id="username"
    placeholder="username"
    type="text"
    value=""
  />
  <input
    type="submit"
    value="search"
  />
</form>
`;
